internal/command/providers_lock.go

```diff
@@ -210,17 +210,18 @@ func (c *ProvidersLockCommand) Run(args []string) int {
 	// Because our Installer abstraction is a per-platform idea, we'll
 	// instantiate one for each of the platforms the user requested, and then
 	// merge all of the generated locks together at the end.
-	updatedLocks := map[getproviders.Platform]*depsfile.Locks{}
-	selectedVersions := map[addrs.Provider]getproviders.Version{}
-	for _, platform := range platforms {
-		tempDir, err := os.MkdirTemp("", "terraform-providers-lock")
+	//
+	// TODO lock UI Output?
```

Comment:
This is a common them across most of the recent performance PR's I've opened.  I don't know if there is any locking on the output streams.